tests/wpunit/Common/Libraries/Uplink/Controller_Test.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace TEC\Common\Libraries\Uplink;
+
+use tad\Codeception\SnapshotAssertions\SnapshotAssertions;
+use TEC\Common\StellarWP\Uplink\Config;
+use TEC\Common\StellarWP\Uplink\Register;
+use TEC\Common\StellarWP\Uplink\Resources\Collection;
+use Tribe__Main;
+use Tribe\Tests\Traits\With_Uopz;
+
+class Controller_Test extends \Codeception\TestCase\WPTestCase {
+	use SnapshotAssertions;
+	use With_Uopz;
+
+	/**
+	 * @var Controller
+	 */
+	private $controller;
+
+	/**
+	 * @before
+	 */
+	public function setup_uplink() {
+		$this->controller = tribe( Controller::class );
+		$this->controller->register_uplink();
```

Comment:
The provider is not being loaded by common's loading process ? I am not sure if it should cause the provider's load is bound in `plugins_loaded` ( @see `common/src/Tribe/Main.php` `tribe_register_provider( Libraries\Provider::class );` )

Reply:
If I remove `$this->controller->register_uplink();` each test fails because the plugin has already been registered.  Setting it up like this allows all of the tests to pass correctly. Otherwise, I get the error `[TEC\Common\StellarWP\Uplink\Exceptions\ResourceAlreadyRegisteredException] The resource "common-test-slug" is already registered.`